modules/webgl/src/adapter/webgl-device.ts

```diff
@@ -128,11 +128,13 @@ export class WebGLDevice extends Device {
 
     this.handle = createBrowserContext(this.canvasContext.canvas, {
       ...props,
-      onContextLost: (event: Event) =>
+      onContextLost: (event: Event) => {
```

Comment:
What are these changes? Some editor auto correct?

Reply:
> This calls the users onContextLost, if provided

I see. Adding a one line bug fix in to a bigger API change is a bit risky, can slip through the review, and can get lost if API change is not approved...

Reply:
Happy to split this out. I didn't mean to put anything risky in here. The reason I included this is get clarity on the intended API. It's unclear to me what the intent is.

If users shouldn't be able to provide `onContextLost` by design, then the API types need to be adjusted a bit. If we can agree the user should be able to provide `onContextLost`, then this is a bug and I'll pull that into another PR.

Reply:
Now that you shine a light on it, I think we should completely drop support for `onContextLost`. Such callback code tends to be surprisingly messy to keep working across devices and it doesn't add value.

We don't talk about contexts anymore, just devices. and we have the replacement in the WebGPU style promise API: `device.lost`

App can register its onContextLost callback with `device.lost.then(onContextLost)`.

Reply:
I agree. `device.lost` sounds better since it the same API for both devices, making this redundant.

Is there a device equivalent for `onContextRestore` as well?

Reply:
No, context restore is not supported by luma.gl v9. 

The problems is that properly handling context restore in non-trivial apps is very tricky, since all resources are lost and have to be recreated, and the data that was used to create buffers and textures may not have been saved. 

\Showing a message asking the user to just refresh the page is usually best way out. 

WebGPU doesn't support restore.

Reply:
Should the onContextRestore code in the WebGL device be removed then?

Reply:
Yes. I would vote for that. It is not impossible for an app to register the callback even if luma.gl API doesn't support it, they will just need to get the context and add an event handler themselves.